### PR TITLE
Improve boilerplate code with tests and other items.

### PR DESCRIPTION
**Pull request by redhug1**
1. Add comments.
2. Add service_test.go and support files (lifted from Florence).
3. Move ctx into main()
4. Move Get() of config into main’s Run() to make testing more flexible.
5. Tweak config settings.
6. Improve config_test.go to achieve 100% code coverage.
7. Improve service.go
8. Added ‘ADD CODE:’ comments that are effective indicators for where someone picking up the boilerplate code should/can make their changes and then remove the ‘ADD CODE:’ comments once their project nears completion.
9. Added (and commented out) in main.go pre-initialised values for BuildTime, GitCommit & Version - which when uncommented to replace the non initialised copies allows one to debug the code is vscode, etc.
10. Update manifest file.

### CODE REVIEW

File: project_generation/content/templates/api/main.go.tmpl [hunk 1]
```diff
@@ -19,23 +20,42 @@ var (
...
 
-func run() error {
+func run(ctx context.Context) error {
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, os.Interrupt, os.Kill)
 
+	// Run the service, providing an error channel for fatal errors
 	svcErrors := make(chan error, 1)
-	svc, err := service.Run(BuildTime, GitCommit, Version, svcErrors)
+	svcList := service.NewServiceList(&service.Init{})
```
Comment (DavidSubiros) [inline project_generation/content/templates/api/main.go.tmpl:47]: Just an observation, maybe for a future task: I think that the service initialiser may be confusing, and we should follow `dp-filter-api` as an example. But I wouldn't block this task for this reason.

File: project_generation/content/templates/api/main.go.tmpl [hunk 2]
```diff
@@ -45,9 +65,7 @@ func run() error {
 	case err := <-svcErrors:
 		return errors.Wrap(err, "service error received")
```
Comment (DavidSubiros) [inline project_generation/content/templates/api/main.go.tmpl:66]: I don't know if we should try to do the graceful shutdown if there is an error too?

File: project_generation/content/templates/api/service/service.go.tmpl [hunk 3]
```diff
@@ -61,32 +61,67 @@ func Run(buildTime, gitCommit, version string, svcErrors chan error) (*Service,
...
+		if svc.ServiceList.HealthCheck {
+			svc.HealthCheck.Stop()
+		}
+
+		// stop any incoming requests before closing any outbound connections
+		if err := svc.Server.Shutdown(ctx); err != nil {
+			log.Event(ctx, "failed to shutdown http server", log.Error(err), log.ERROR)
+			hasShutdownError = true
+		}
+
```
Comment (DavidSubiros) [inline project_generation/content/templates/api/service/service.go.tmpl:93]: I think we should have a comment here, like:
```
ADD CODE HERE: Close other dependencies, in the expected order
```
or something similar.

**Review by DavidSubiros — COMMENTED** [review]
The changes look good! I just added minor comments, but maybe someone else should check it? Also, I think that the initialiser could be removed, and then the dependency initialisation could be simplified (similar to dp-filter-api)

File: project_generation/content/templates/api/main.go.tmpl (hunk 2 above)
Comment (redhug1) [reply project_generation/content/templates/api/main.go.tmpl:66]: leave this one for another PR (maybe)

File: project_generation/content/templates/api/main.go.tmpl (hunk 1 above)
Comment (redhug1) [reply project_generation/content/templates/api/main.go.tmpl:47]: leave this one for another PR (maybe)

File: project_generation/content/templates/api/main.go.tmpl (hunk 2 above)
Comment (redhug1) [reply project_generation/content/templates/api/main.go.tmpl:66]: OK, hint added

File: project_generation/content/templates/api/service/service.go.tmpl (hunk 3 above)
Comment (redhug1) [reply project_generation/content/templates/api/service/service.go.tmpl:93]: ok, done

**Review by DavidSubiros — APPROVED** [review]

**Pull request merged**